thirdeye/thirdeye-frontend/app/pods/rootcause-anomalyfunction-cache/service.js

```diff
@@ -0,0 +1,126 @@
+import Service from '@ember/service';
+import {
+  toFilters,
+  toFilterMap
+} from 'thirdeye-frontend/utils/rca-utils';
+import { checkStatus } from 'thirdeye-frontend/utils/utils';
+import fetch from 'fetch';
+import _ from 'lodash';
+import moment from 'moment';
+
+export default Service.extend({
+  timeseries: null, // {}
+
+  context: null, // {}
+
+  pending: null, // Set
+
+  errors: null, // Set({ urn, error })
+
+  init() {
+    this.setProperties({ timeseries: {}, context: {}, pending: new Set(), errors: new Set() });
+  },
+
+  clearErrors() {
+    this.setProperties({ errors: new Set() });
+  },
+
+  request(requestContext, urns) {
+    const { context, timeseries, pending } = this.getProperties('context', 'timeseries', 'pending');
+
+    const metrics = [...urns].filter(urn => urn.startsWith('frontend:anomalyfunction:'));
+
+    // TODO eviction on cache size limit
```

Comment:
Question: what's the limit and will exceeding it simply mean no new data is cached?

Reply:
the aren't any cache limits atm, however, the cache currently gets flushed any time the user changes the investigation period or -metric